src/frankenphp-symfony/src/Runner.php

```diff
@@ -56,8 +57,40 @@ public function run(): int
             }
 
             gc_collect_cycles();
-        } while ($ret && (-1 === $this->loopMax || ++$loops < $this->loopMax));
+            ++$loops;
+        } while ($ret && (-1 === $this->loopMax || $loops < $this->loopMax) && $this->hasAvailableMemory($loops));
 
         return 0;
     }
+
+    private function hasAvailableMemory(int $loops): bool
+    {
+        if (-1 === $this->minMemoryMb || 1 === $loops) { // Always allow the first cycle
```

Comment:
If `1 === $loops`, then we have completed the first request. If we are out of memory after the first request, we should restart, right? 

I suggest removing this check.